tricycle_controller/CMakeLists.txt

```diff
@@ -91,29 +74,46 @@ if(BUILD_TESTING)
     tf2_msgs
   )
 
-  ament_add_gmock(
-    test_load_tricycle_controller
+  ament_add_gmock(test_load_tricycle_controller
     test/test_load_tricycle_controller.cpp
   )
-  target_include_directories(test_load_tricycle_controller PRIVATE include)
+  target_link_libraries(test_load_tricycle_controller
+    tricycle_controller
+  )
   ament_target_dependencies(test_load_tricycle_controller
     controller_manager
     ros2_control_test_assets
   )
```

Comment:
You can specify the visibility here. You seem to be doing that everywhere else.

Reply:
you can't specify visibility if you use the ament_add_gtest function because it does a target_link_libraries internally without visibility

Reply:
Reference to the issue so others know about this: https://github.com/ament/ament_cmake/issues/405

Reply:
Thanks!